src/ray/core_worker/transport/out_of_order_actor_scheduling_queue.h

```diff
@@ -84,6 +90,10 @@ class OutOfOrderActorSchedulingQueue : public SchedulingQueue {
   /// Whether we should enqueue requests into asyncio pool. Setting this to true
   /// will instantiate all tasks as fibers that can be yielded.
   bool is_asyncio_ = false;
+  /// Mutext to protect attributes used for thread safe APIs.
+  absl::Mutex mu_;
+  /// A map of actor task IDs -> is_canceled
+  absl::flat_hash_map<TaskID, bool> pending_tasks_queued_or_executing GUARDED_BY(mu_);
```

Comment:
ditto

Reply:
TODO: seems we are implementing similar code in out_of_order_actor_scheduling_queue and actor_scheduing_queue, it'd be nice to unify them.

Reply:
Added to TODO